adapters/repos/db/vector/hnsw/search.go

```diff
@@ -285,14 +371,8 @@ func (h *hnsw) searchLayerByVectorWithDistancer(queryVector []float32,
 
 			if distance < worstResultDistance || results.Len() < ef {
 				candidates.Insert(neighborID, distance)
-				if level == 0 && allowList != nil {
-					// we are on the lowest level containing the actual candidates and we
-					// have an allow list (i.e. the user has probably set some sort of a
-					// filter restricting this search further. As a result we have to
-					// ignore items not on the list
-					if !allowList.Contains(neighborID) {
-						continue
-					}
+				if !h.acornSearch.Load() && allowList != nil && !allowList.Contains(neighborID) {
```

Comment:
This changes original sweep functionality of only skipping for level = 0. Can we add back the level == 0 check?